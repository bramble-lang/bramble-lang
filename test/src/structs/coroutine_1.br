
fn my_main( p: bool) {
    let ms:MyStruct := MyStruct {x: 5, y: 13, z: true};
    let c:i32 := init my_co(ms, ms);
    let w : i32 := yield c;
    let x : i32 := 1;
    printiln x;
    printiln w;
    let z : i32 := yield c;
    printiln z;
    return;
}

co my_co (x: MyStruct, y: MyStruct) -> i32 {
    yret x.y + y.y;
    return x.x;
}

struct MyStruct {
    x: i32,
    y: i32,
    z: bool,
}