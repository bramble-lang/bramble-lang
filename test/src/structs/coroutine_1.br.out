5
1
26
13
25
